src/traces/scatter/fillcolor_defaults.js

```diff
@@ -18,9 +27,25 @@ module.exports = function fillColorDefaults(traceIn, traceOut, defaultColor, coe
         }
     }
 
+    var averageGradientColor;
+    if(traceIn.fillgradient) {
```

Comment:
Let's remove this if statement and coerce `fillgradient` attributes.
Please note that `fillgradient` could also be set via `template`. In that case `traceIn.fillgradient` could be undefined. 
For that reason we should try to always look into the `traceOut` not the `traceIn`.